py2many/clike.py

```diff
@@ -513,7 +515,7 @@ def visit_AnnAssign(self, node):
         return (target, type_str, val)
 
     def visit_Delete(self, node):
-        raise AstNotImplementedError("del not implemented", node)
+        return "Del /* Unimplemented */"
```

Comment:
why? this would be a regression towards the goal of https://github.com/adsharma/py2many/issues/291

Reply:
I think you want `visit_unsupported_body()` which results in a runtime assertion failure in the target language.
Once we raise AstNotImplementedError(), all subsequent code is not transpiled.

We could have a flag that controls whether `visit_unsupported_body()` throws or generates runtime assertion failure.

Reply:
Also we have only two instances of these:

```
del target.elts[i]
```

This could transpile to the following in cpp:

```
target_elts.erase(target_elts.begin() + i)
```

Reply:
No.  The unsupported node raises an exception at that node.  Nodes which can represent unsupported nodes then catch that exception and generate an alternative representation if the user asked for it.  The nodes which can easily do this are functions, classes, etc.  Assignment nodes will eventually be able to do it right.

Reply:
That's different from what you wrote in #291. The problem that this code is trying to fix is the following pattern:

```
if condition:
   body-with-unsupported
else:
  body-with-supported
```

Then:

```
def visit_If(...):
      ...
      return "\n".join(buf)
```

The unsupported-body transpiles to `None` (because `visit_$Construct` was missing) and then when we try to join it, things break.

I'm not convinced that raising an exception for unsupported is the right answer for all cases. For example, even if you catch it somewhere else, you'll never visit the `else` part of the expression above won't uncover further transpilation problems. It's not clear that it's 1 or 2 places where the exception needs to be caught.

I don't want to hold this work while we explore a solution to that problem. Let's use #291 to discuss it. Returning a string fixes the problem with join() and allows us to make forward progress. We can make the behavior controlled by a flag, return a different string etc if it makes sense.